### DISCUSSION TITLE
Dokumentation von KitodoScripts

### DISCUSSION THREAD

**Discussion by aetherfaerber** [kitodo/kitodo-production#6004] [Q&A] [answered]
Wir sind bei der Suche nach Lösungen zu unseren Problemen (z.B. Löschen vieler Vorgänge oder Aufgabenstatus für Vorgänge setzen) wiederholt auf vorhandene KitodoScripts gestoßen.
Leider konnten wir aber dazu keine Dokumentation finden. Sowohl dazu, welche es gibt, wie sie an die Anwendung angeschlossen sind oder wie sie zu bedienen könnt. [Ein Pull-Request zur besseren Sichtbarkeit](https://github.com/kitodo/kitodo-production/pull/4462) gibt nur geringe Aufschlüsse.

Gibt es eine übergreifende Dokumentation zu diesen Funktionen oder Anwenungsszenarien, die dokumentiert wurden?

Weißt du etwas, @matthiaskissler?

**Comment by henning-gerhardt** [comment, marked as answer]
Ein wenig zu den KitodoScripts kann man im Wiki unter 
- https://github.com/kitodo/kitodo-production/wiki/KitodoScript
- https://github.com/kitodo/kitodo-production/wiki/Metadaten-Skripte

**Reply by aetherfaerber** [reply]
Vielen Dank! Die beiden Seiten waren mir bekannt, allerdings war ich bislang nicht sicher, ob sie beide dasselbe Feature beschreiben. Damit ergibt sich insgesamt schon ein etwas umfassenderes Bild, das aber noch Lücken hat:

* die Wiki-Seite „KitodoScript“ enthält mehrere Einträge, deren Erläuterung nur „Zu diesem Befehl liegen keine gesicherten Informationen vor.“ enthält.
* auf der Seite „Metadaten-Skripte“ habe ich gerade einen kleinen Fehler behoben.

Für die Ergänzung der Beschreibungen dürften erst einige Tests notwendig sein, um herauszufinden, was denn die Funktion der Skripte ist.

Schon jetzt gäbe es m.E. zwei Schritte, die die Sache ohne weiteres handhabbarer machen würden:
* die beiden genannten Seiten miteinander verlinken
* nicht nur auf die manuelle Bedienung über die Oberfläche verweisen, sondern auch die im oben genannten Issue gezeigten Möglichkeit zur Steuerung im Workflow erklären
